src/server/worker/pipeline/transform/registry.go

```diff
@@ -461,10 +461,21 @@ func deserializeDatumSet(any *types.Any) (*DatumSet, error) {
 }
 
 func (reg *registry) processJobEgressing(pj *pendingJob) error {
-	url := pj.ji.Details.Egress.URL
-	err := pj.driver.PachClient().GetFileURL(pj.commitInfo.Commit, "/", url)
-	// file not found means the commit is empty, nothing to egress
-	if err != nil && !pfsserver.IsFileNotFoundErr(err) {
+	client := pj.driver.PachClient()
+	egress := pj.ji.Details.Egress
+	var request pfs.EgressRequest
+	request.Commit = pj.commitInfo.Commit
+	if pj.ji.Details.Egress.URL != "" {
+		request.Target = &pfs.EgressRequest_Obj{Obj: egress.GetObj()}
+	} else {
+		switch egress.Target.(type) {
+		case *pps.Egress_Obj:
+			request.Target = &pfs.EgressRequest_Obj{Obj: egress.GetObj()}
+		case *pps.Egress_Sql:
+			request.Target = &pfs.EgressRequest_Sql{Sql: egress.GetSql()}
+		}
+	}
+	if _, err := client.Egress(client.Ctx(), &request); err != nil && !pfsserver.IsFileNotFoundErr(err) {
```

Comment:
maybe put a TODO here around detecting unrecoverable errors in the SQL case (bad password, datatype mismatch)